src/v/cluster/tests/archival_metadata_stm_test.cc

```diff
@@ -453,3 +454,35 @@ FIXTURE_TEST(
 
     archival_stm.stop().get();
 }
+
+FIXTURE_TEST(test_archival_stm_batching, archival_metadata_stm_fixture) {
+    wait_for_confirmed_leader();
+    std::vector<cloud_storage::segment_meta> m;
+    m.push_back(segment_meta{
+      .base_offset = model::offset(0),
+      .committed_offset = model::offset(999),
+      .archiver_term = model::term_id(1),
+      .segment_term = model::term_id(1)});
+    m.push_back(segment_meta{
+      .base_offset = model::offset(1000),
+      .committed_offset = model::offset(1999),
+      .archiver_term = model::term_id(1),
+      .segment_term = model::term_id(1)});
+    m.push_back(segment_meta{
+      .base_offset = model::offset(0),
+      .committed_offset = model::offset(999),
+      .archiver_term = model::term_id(2),
+      .segment_term = model::term_id(1)});
+    // Replicate add_segment_cmd command that adds segment with offset 0
+    auto batcher = archival_stm->batch_start(ss::lowres_clock::now() + 10s);
+    batcher.add_segments(m);
+    batcher.cleanup_metadata();
+    batcher.replicate().get();
+    archival_stm->sync(10s).get();
```

Comment:
nit: why do we need to call sync here? I think `replicate()` waits for the commands to be applied, no?

Reply:
fixed